components/tools/rag-details.tsx

```
@@ -0,0 +1,159 @@
+                <div className="flex flex-col gap-4">
+                  <div className="flex flex-col gap-2">
+                    <div className="text-lg font-medium">Question</div>
+                    <p className="text-muted-foreground">{args?.question}</p>
+                  </div>
+                  <div className="flex flex-col gap-2">
+                    <div className="text-lg font-medium">Results</div>
+                    {isLoading && (
+                      <span className="text-muted-foreground">Getting results...</span>
+                    )}
+                    {result?.map((item: any, index: number) => (
+                      <div key={index} className="flex flex-col gap-2 relative">
```

Biome (1.9.4)

[error] Avoid using the index of an array as key property in an element.

This is the source of the key value.

The order of the items may change, and this also affects performances and component state.
Check the React documentation. 

(lint/suspicious/noArrayIndexKey)